lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/clazz/inviteViaContact/InviteViaContactScreen.kt

```diff
@@ -0,0 +1,147 @@
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.dokar.chiptextfield.Chip
+import com.dokar.chiptextfield.m3.OutlinedChipTextField
+import com.dokar.chiptextfield.rememberChipTextFieldState
+import com.ustadmobile.core.MR
+import com.ustadmobile.core.viewmodel.clazz.inviteviaContact.InviteViaContactUiState
+import com.ustadmobile.core.viewmodel.clazz.inviteviaContact.InviteViaContactViewModel
+import com.ustadmobile.libuicompose.components.UstadContactPickButton
+import dev.icerock.moko.resources.compose.stringResource
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.input.ImeAction
+import com.dokar.chiptextfield.m3.ChipTextFieldDefaults
+
+
+@Composable
+fun InviteViaContactScreen(
+    viewModel: InviteViaContactViewModel
+) {
+
+    val uiState by viewModel.uiState.collectAsState(InviteViaContactUiState())
+    InviteViaContactScreen(
+        uiState = uiState,
+        onChipSubmitClick = {
+            viewModel.onClickChipSubmit(
+                it
+            )
+        }
+
+    )
+}
+
+class AvatarChip(text: String, val avatar: ImageVector) : Chip(text)
+
+@Composable
+fun InviteViaContactScreen(
+    uiState: InviteViaContactUiState = InviteViaContactUiState(),
+    onChipSubmitClick: (String) -> Unit,
+) {
+    val state = rememberChipTextFieldState<AvatarChip>()
+    val scrollState = rememberScrollState()
+    Column(
+        modifier = Modifier
+            .fillMaxSize().verticalScroll(scrollState), // Make the column scrollable
+
+    ) {
+
+
+        OutlinedChipTextField(
+            state = state,
+            modifier = Modifier
+                .weight(1f)
+                .padding(10.dp).fillMaxWidth(),
+
+            onSubmit = {
+
+                onChipSubmitClick(it)
+
+                null
+
+            },
+
+            chipStyle = ChipTextFieldDefaults.chipStyle(shape = RoundedCornerShape(20.dp)),
+            textStyle = TextStyle(fontSize = 13.sp),
+            chipLeadingIcon = { chip -> Avatar(chip, modifier = Modifier) },
+            label = { Text(text = stringResource(MR.strings.add_username_email_phone)) },
+            chipHorizontalSpacing = 8.dp,
+            chipVerticalSpacing = 10.dp,
+            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next),
+        )
+
+        UstadContactPickButton(
+            onContactPicked = {
+                if (it != null) {
+                    onChipSubmitClick(it)
+                }
+            }
+        )
+    }
+
+    LaunchedEffect(uiState.chips) {
```

Comment:
I don't think this function will work as intended. Every time the uiState chips change it will add it again

e.g. user adds chip A, goes to view model, comes back on ui State, gets added to chip state [uistate = [A], chip state = [A].
user adds chip B, goes to view model, come back on ui state as [A,B], both get added, now chip state [ A, A, B]

Reply:
This needs to check the material chips state to see if the given chip is already there before adding it.

Reply:
Also I don't see code handling the removal.

Reply:
we dont neet to worry about removal of chips because when we remove any chip this state = rememberChipTextFieldState<AvatarChip>()  will remove chip on it own

Reply:
Yes we do need to worry about removal: when the user actually goes and clicks invite, then the viewmodel needs to know (otherwise the invite request will include items removed by the user).

Another reason that state needs to be in the ViewModel: we use the saved state handle (in the view model) e.g. if the app is destroyed when the user leaves due to inactivity, and then returns, we should restore the state.